core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -244,4 +245,41 @@ private TableProperties() {
 
   public static final String UPSERT_MODE_ENABLE = "write.upsert.enable";
   public static final boolean UPSERT_MODE_ENABLE_DEFAULT = false;
+
+  /**
+   * Get a table property for a specific Iceberg action's config option.
+   * <p>
+   * For example, {@link org.apache.iceberg.actions.RewriteDataFiles#TARGET_FILE_SIZE_BYTES} corresponds to
+   * table property actions.rewrite-data-files.target-file-size-bytes
+   *
+   * @param action name of the iceberg action, retrieved from the NAME variable of the specific action if exists.
+   *               The name is assumed to be in lower-case-kebab-style.
+   * @param option config option name, typically defined as public static variables in the specific action.
+   *               The name is assumed to be in lower-case-kebab-style.
+   * @return table property name
+   */
+  public static String actionTableProperty(String action, String option) {
+    return String.format("actions.%s.%s", action, option);
+  }
+
+  /**
+   * Get a table property for a specific Iceberg action and strategy's config option.
+   * <p>
+   * For example, {@link org.apache.iceberg.actions.BinPackStrategy#MIN_INPUT_FILES} corresponds to
+   * table property actions.rewrite-data-files.strategies.binpack.min-input-files
+   *
+   * @param action name of the iceberg action, retrieved from the NAME variable of the specific action if exists.
+   *               The name is assumed to be in lower-case-kebab-style.
+   * @param strategy name of the strategy, retrieved from the name method of the specific strategy,
+   *                 such as {@link org.apache.iceberg.actions.RewriteStrategy#name()}.
+   *                 The name is assumed to be in CAPITALIZED_CASE_SNAKE_STYLE, and is converted to
+   *                 lower-case-kebab-style in the config key.
+   * @param option config option name, typically defined as public static variables in the specific strategy.
+   *               The name is assumed to be in lower-case-kebab-style.
+   * @return table property name
+   */
+  public static String actionStrategyTableProperty(String action, String strategy, String option) {
```

Comment:
Why make these strategy properties? Aren't they configured on the action itself?

Reply:
Yes that's right. The intention here is to allow users to be able to find the option, as the option itself is defined in the strategy class instead of the action class. For example, when seeing `actions.rewrite-data-files.strategies.binpack.min-input-files`, users immediately know the option is defined in the `BinPackStrategy` class. But I am also fine with not having that, no strong preference here.

Reply:
I think that the idea behind using properties is that users don't need to know which properties are defined by the strategy or which are defined by the action. The error message is "not supported by the action or the strategy". So let's just keep this to `actions.<action-name>.<option>` to keep it simple.

Reply:
sounds good, I will update it that way.